lib/aws-xray-sdk/lambda/facade_segment.rb

```diff
@@ -0,0 +1,13 @@
+module XRay
+  class FacadeSegment < XRay::Segment
```

Comment:
FacadeSegment should restrict any mutation to itself except adding/removing subsegments and should raise `UnsupportedOperation` exception. Can you add those methods too? Please refer to the [python SDK FacadeSegment](https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/facade_segment.py) for details.

Reply:
Done